app/controllers/sign_in/client_configs_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module SignIn
+  class ClientConfigsController < SignIn::ApplicationController
+    rescue_from ActiveRecord::RecordNotFound, with: :not_found
+    skip_before_action :authenticate
+    before_action :authenticate_service_account
+    before_action :set_client_config, only: %i[show update destroy]
+
+    def index
+      client_configs = SignIn::ClientConfig.where(client_id: params[:client_ids])
+
+      render json: client_configs, status: :ok
+    end
+
+    def show
+      render json: @client_config, status: :ok
+    end
+
+    def create
+      client_config = SignIn::ClientConfig.new(client_config_params)
+
+      if client_config.save
+        render json: client_config, status: :created
+      else
+        render json: client_config.errors, status: :unprocessable_entity
+      end
+    end
+
+    def update
+      if @client_config.update(client_config_params)
+        render json: @client_config, status: :ok
+      else
+        render json: @client_config.errors, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      if @client_config.destroy
+        head :no_content
+      else
+        render json: @client_config.errors, status: :unprocessable_entity
+      end
+    end
+
+    private
+
+    def client_config_params
+      params.require(:client_config).permit(:client_id, :authentication, :anti_csrf, :redirect_uri,
+                                            :access_token_duration, :access_token_audience, :refresh_token_duration,
+                                            :logout_redirect_uri, :pkce, :refresh_token_path, certificates: [])
+    end
+
+    def set_client_config
+      @client_config = SignIn::ClientConfig.find(params[:id])
```

Comment:
I think we want to do this find by the client_id

Reply:
Or do you think there's an advantage to querying by the internal id versus client_id

Reply:
I originally had it that way but in order for update and destroy to work with `ActiveResource` it needs to be `id`. Perhaps I can make `show` use `client_id` and `update` and `destroy` use `id`. I am making it work by just passing a single `client_id` to index

Reply:
I guess I can live with it. Since it's a backend only api it won't reveal info about the schema. Typically I don't like passing primary keys around but that's more a concern for apis with non-secure clients